api/server.js

```diff
@@ -41,36 +54,106 @@ const DB_NAME = "huskyReads"; // Database name
  * Login endpoint
  */
 app.post("/login", async (req, res) => {
-  try {
-    const db = await getDBConnection();
-    await db.close();
-  } catch (err) {
-    loggingModule(err, "login");
-  }
+	try {
+		res.type("text");
+		let username = req.body.username;
+		let password = req.body.password;
+		if (!username || !password) {
+			res.status(CLIENT_ERROR_CODE_400).send("Missing username or password");
+		} else {
+			let returns = await getPassword(username);
+			if (returns.length >= 1 && password === returns[0]["password"]) {
+				res.status(SUCCESS_CODE).send("Login Successful");
+			} else {
+				res.status(CLIENT_ERROR_CODE_401).send("Invalid user credentials");
+			}
+		}
+	} catch (err) {
+		loggingModule(err, "login");
+		res.status(SERVER_ERROR_CODE).send(SERVER_ERROR_MESSAGE);
+	}
 });
 
 /**
  * Sign up endpoint (Create new User)
  */
 app.post("/signup", async (req, res) => {
-  try {
-    const db = await getDBConnection();
-    await db.close();
-  } catch (err) {
-    loggingModule(err, "signup");
-  }
+	try {
+		res.type("text");
+		let username = req.body.username;
+		let password = req.body.password;
+		if (!username || !password) {
+			res.status(CLIENT_ERROR_CODE_400).send("Missing username or password");
+		} else if (await checkIfExist(username)) {
+			res.status(CLIENT_ERROR_CODE_400).send("Username already taken");
+		} else {
+			let info = [username, password];
+			createUser(info);
+			res.status(SUCCESS_CODE).send("Signup Successful");
+		}
+	} catch (err) {
+		loggingModule(err, "signup");
+		res.status(SERVER_ERROR_CODE).send(SERVER_ERROR_MESSAGE);
+	}
+});
+
+/**
+ * Update user color scheme
+ */
+app.post("/color_scheme", async (req, res) => {
+	try {
+		res.type("text");
+		let username = req.body.username;
+		let color_scheme = req.body.color_scheme;
+		if (!username || !color_scheme) {
+			res.status(CLIENT_ERROR_CODE_400).send("Missing username or color_scheme");
+		} else if (await !checkIfExist(usernamme)) { 
+			res.status(CLIENT_ERROR_CODE_401).send("Invalid Username");
+		} else {
+			let info = [username, color_scheme];
+			if (!checkColor) {
```

Comment:
The `checkColor` method isn't called here correctly. Must add parentheses with `color_scheme` parameter